src/codegen/mod.rs

```diff
@@ -1224,6 +1224,79 @@ impl MethodCodegen for Method {
     }
 }
 
+enum EnumBuilder<'a> {
```

Comment:
Nitpick: it would be nice to have brief documentation comments here and throughout the new code (I know I never got around to documenting the codegen module, but we should aim to hold ourselves to documenting new code)